packages/compass-collection/src/components/collection-tab.tsx

```diff
@@ -57,12 +62,35 @@ const collectionModalContainerStyles = css({
 });
 
 const CollectionTab: React.FunctionComponent<{
+  localAppRegistry: AppRegistry | undefined;
   currentTab: string;
+  collectionTabPluginMetadata: CollectionTabPluginMetadata;
   renderScopedModals(): React.ReactElement[];
   renderTabs(): { name: string; component: React.ReactElement }[];
   onTabClick(name: string): void;
-}> = ({ currentTab, renderScopedModals, renderTabs, onTabClick }) => {
-  const tabs = renderTabs();
+}> = ({
+  localAppRegistry,
+  currentTab,
+  collectionTabPluginMetadata,
+  renderScopedModals,
+  renderTabs,
+  onTabClick,
+}) => {
+  const pluginTabs = useCollectionTabPlugins();
+  const legacyTabs = renderTabs();
+  const tabs = [
+    ...legacyTabs,
+    ...(localAppRegistry ? pluginTabs : []).map(
+      ({ name, component: Component }) => ({
+        name,
+        component: (
+          <AppRegistryProvider localAppRegistry={localAppRegistry}>
+            <Component {...collectionTabPluginMetadata} />
+          </AppRegistryProvider>
```

Comment:
@gribnoysup ^ this probably looks wrong in some way? Wanted to get something minimal that seems to work fine to avoid conflicts with COMPASS-7402

Reply:
Yeah, the scope for the tab should be controlled by the parent, not the tab itself, this is why this localAppRegistry state is already part of the store in the tabs store, not in the collection-tab store

Reply:
@gribnoysup How does 92c0d6afbdcee14fc104dbe33c024469fa3c4715 look to you?

Reply:
Actually … with that change, something odd that’s happening is that the top-level tab titles don’t update if I switch to another tab (Indexes is the one in the failing e2e test here) if and only if I open a collection through the collections list view, doesn’t happen when I do it through the sidebar … I’ll probably figure it out, but if this rings a bell to you, lmk :)

Reply:
Ah nvm, `deactivateOnUnmount` did the trick I think :) But tbh I’m a bit surprised that _that_ is where the visible difference between having the option and not having it falls …

Reply:
This state does rely on localAppRegistry events at the moment 😞, as you as you switch the tab, this subscription was lost forever (a less immediately obvious thing was that scoped modals like explain plan would stop working after the first tab was opened and switched)